orderer/common/cluster/clusterservice.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright IBM Corp. 2017 All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cluster
+
+import (
+	"bytes"
+	"encoding/asn1"
+	"fmt"
+	"io"
+	"strconv"
+	"time"
+
+	"github.com/hyperledger/fabric-protos-go/orderer"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/common/util"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+//go:generate mockery -dir . -name ClusterStepStream -case underscore -output ./mocks/
+
+// ClusterStepStream defines the gRPC stream for sending
+// transactions, and receiving corresponding responses
+type ClusterStepStream interface {
+	Send(response *orderer.ClusterNodeServiceStepResponse) error
+	Recv() (*orderer.ClusterNodeServiceStepRequest, error)
+	grpc.ServerStream
+}
+
+// ClusterService defines the raft Service
+type ClusterService struct {
+	StreamCountReporter              *StreamCountReporter
+	Dispatcher                       Dispatcher
+	Logger                           *flogging.FabricLogger
+	StepLogger                       *flogging.FabricLogger
+	MinimumExpirationWarningInterval time.Duration
+	CertExpWarningThreshold          time.Duration
+	NodesCert                        map[uint64][]byte
```

Comment:
you actually need a mapping from channel to map[uint64][]byte, since this service is global for all channels.